jscomp/runtime/typed_array.ml

```diff
@@ -17,48 +17,329 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
+type array_buffer
+type 'a array_like (* should be shared with js_array *)
+
+module type Type = sig
+  type t
+end
+
+
+module ArrayBuffer = struct
+  type t = array_buffer
+
+  external make : int -> t = "ArrayBuffer" [@@bs.new]
+  (** takes length. initializes elements to 0 *)
+
+  (* ArrayBuffer.isView: seems pointless with a type system *)
+  (* experimental
+  external transfer : array_buffer -> t = "ArrayBuffer.transfer" [@@bs.val]
+  external transferWithLength : array_buffer -> int -> t = "ArrayBuffer.transfer" [@@bs.val]
+  *)
+
+  external byteLength : t -> int = "" [@@bs.get]
+
+  external slice : start:int -> end_:int -> array_buffer = "" [@@bs.send.pipe: t]
+  external sliceFrom : int -> array_buffer = "slice" [@@bs.send.pipe: t]
+end
+
+
+module TypedArray (Type: Type) = struct
+  type elt = Type.t
+  type 'a typed_array
+
+  external get : elt typed_array -> int -> elt  = "" [@@bs.get_index]
+  external set : elt typed_array -> int -> elt -> unit = "" [@@bs.set_index]
```

Comment:
Well, I don't know if you need a functor, but I do :) I don't know any other way of inheriting an implementation. The alternative (for me) would be copy-and-paste.